File: 345-ReverseVowelsofaString/345-ReverseVowelsofaString.java
// Last updated: 6/16/2026, 2:15:08 PM
1class Solution {
2
3    HashSet<String> set = new HashSet<>();
4
5    public boolean wordBreak(String s, List<String> wordDict) {
6
7        if(s.length() == 0)
8            return true;
9
10        if(set.contains(s))
11            return false;
12
13        for(int i = 0; i < wordDict.size(); i++) {
14
15            String word = wordDict.get(i);
16
17            if(s.startsWith(word)) {
18
19                if(wordBreak(s.substring(word.length()), wordDict))
20                    return true;
21            }
22        }
23
24        set.add(s);
25        return false;
26    }
27}

File: 345-ReverseVowelsofaString/345-ReverseVowelsofaString.py
# Last updated: 6/22/2026, 1:58:13 PM
1class Solution(object):
2    def reverseVowels(self, s):
3        v = [c for c in s if c in "aeiouAEIOU"][::-1]
4        j = 0
5        res = ""
6
7        for c in s:
8            if c in "aeiouAEIOU":
9                res += v[j]
10                j += 1
11            else:
12                res += c
13
14        return res